src/ngraph/op/util/broadcast_base.cpp

```diff
@@ -0,0 +1,283 @@
+                                          arg_shape[i],
+                                          ". Got ",
+                                          target_shape[axes_mapping_val[i]]);
+                }
+            }
+        }
+    }
+    else if (m_broadcast_spec.m_type == AutoBroadcastType::NUMPY ||
+             m_broadcast_spec.m_type == AutoBroadcastType::PDPD)
+    {
+        if (get_input_partial_shape(0).is_static() && get_input_partial_shape(1).is_static())
+        {
+            auto arg_shape = get_input_shape(0);
+
+            if (shape_constant)
+            {
+                const auto target_shape = shape_constant->get_shape_val();
+                auto start_axis = (m_broadcast_spec.m_type == AutoBroadcastType::PDPD)
+                                      ? m_broadcast_spec.m_axis
+                                      : target_shape.size() - arg_shape.size();
+                NODE_VALIDATION_CHECK(this,
+                                      start_axis >= 0,
+                                      "Broadcast target_shape has smaller rank ",
+                                      target_shape.size(),
+                                      " than arg shape ",
+                                      arg_shape.size());
+                for (auto i = start_axis; i < target_shape.size(); i++)
+                {
+                    NODE_VALIDATION_CHECK(
+                        this,
+                        arg_shape[i - start_axis] == 1 || target_shape[i] == 1 ||
+                            arg_shape[i - start_axis] == target_shape[i],
+                        "Broadcast incorrect target shape. Expecting either 1 or ",
+                        arg_shape[i - start_axis],
+                        " . Got ",
+                        target_shape[i]);
+                    result_shape[i] = std::max(arg_shape[i - start_axis], target_shape[i]);
+                }
+            }
+        }
+    }
+
+    set_input_is_relevant_to_shape(0); // arg - Result element type
+    set_input_is_relevant_to_shape(1); // target_shape - Result shape
+    set_input_is_relevant_to_shape(2); // axes_mapping - Broadcast type
+    set_output_type(0, get_input_element_type(0), result_shape);
+}
+
+std::pair<bool, AxisSet> op::util::BroadcastBase::get_broadcast_axes() const
+{
+    AxisSet broadcast_axes;
+    bool axes_known = false;
+
+    if (m_broadcast_spec.m_type == AutoBroadcastType::NONE)
+    {
+        const auto axes_mapping_constant =
+            as_type_ptr<op::v0::Constant>(input_value(2).get_node_shared_ptr());
+        if (get_input_partial_shape(1).is_static() && axes_mapping_constant)
+        {
+            auto target_shape = get_input_shape(1);
+            NGRAPH_CHECK(target_shape.size() == 1);
+            auto axes_mapping_val = axes_mapping_constant->get_axis_vector_val();
+
+            std::vector<size_t> axes(target_shape[0]);
+            std::iota(axes.begin(), axes.end(), 0);
```

Comment:
I got a compilation failure on mac:

/Users/elazarev/Documents/git/ie_dldt/ngraph/src/ngraph/op/util/broadcast_base.cpp:223:18: error: no member named 'iota' in namespace 'std'
            std::iota(axes.begin(), axes.end(), 0);

Reply:
fixed